python/lsst/ip/diffim/modelPsfMatch.py

```diff
@@ -372,15 +394,42 @@ def _buildCellSet(self, exposure, referencePsfModel):
                 widthList.append(widthS)
                 heightList.append(heightS)
 
-        lenMax = max(max(heightList), max(widthList))
-        lenMin = min(min(heightList), min(widthList))
+        psfSize = max(max(heightList), max(widthList))
 
-        lenPsfScience = lenMax if self.config.padPsf else lenMin
-        dimenS = afwGeom.Extent2I(lenPsfScience, lenPsfScience)
+        if self.config.doAutoPadPsf:
+            minPsfSize = nextOddInteger(self.kConfig.kernelSize*self.config.autoPadPsfTo)
+            paddingPix = max(0, minPsfSize - psfSize)
+        else:
+            if self.config.padPsfBy % 2 != 0:
+                raise ValueError("Config padPsfBy (%i pixels) must be even number." %
```

Comment:
I considered that.  My reasoning for raising an error is that when setting doAutoPadPsf=False, I presume the user has reasons for turning off the automatic tuning,  and thus wants exact manual control over the configs and wants ModelPsfMatchTask to just do exactly what they ask of it or quit. The behavior is consistent:  Kernel too big?  Raise. Padding inappropriately odd? Raise. I suspect this config option will only be used when users want precise control such as for testing config parameters like you did for the HSC image